src/main/java/de/muenchen/mobidam/security/CodeDetectionProcessor.java

```diff
@@ -45,8 +48,9 @@ public void process(Exchange exchange) throws Exception {
         if (!mobilithekInterface.getMaliciousCodeDetectionEnabled()) {
             return;
         }
-        InputStreamCache stream = exchange.getIn().getBody(InputStreamCache.class);
-        stream.reset();
+        StreamCache receivedStream = exchange.getIn().getBody(StreamCache.class);
+        receivedStream.reset();
+        InputStream stream = receivedStream instanceof InputStreamCache ? (InputStreamCache) receivedStream : (FileInputStreamCache) receivedStream;
```

Comment:
Achtung!

Reply:
Je nach Mobilitheks Ressource die wir ansprechen bekommen wir ein InputStreamCache oder FileInputStreamCache. 
Mit einem exchange.getMessage().getBody(StreamCache.class); bekommen wir aus dem Body in beiden Fällen etwas zurück. Aber die Benutzung von instanceof ist zu kompliziert gedacht. Das lässt sich vereinfachen zu :
 InputStream stream = (InputStream)receivedStream;